src/ConfigAggregator.php

```diff
@@ -92,6 +96,34 @@ public function getMergedConfig()
         return $this->config;
     }
 
+    /**
+     * Throw an exception if duplicate config providers are found.
+     *
+     * @param ProviderIterable $providers
+     * @return ProviderIterable
+     * @throws InvalidConfigProviderException
+     */
+    private function validateNoDuplicateProviders(iterable $providers): iterable
+    {
+        $uniqueProviders = [];
+
+        foreach ($providers as $provider) {
+            if (! is_string($provider) && ! is_object($provider)) {
+                continue;
```

Comment:
Because you are returning `$uniqueProviders`, this method is also effectively filtering entirely invalid entries too, so `[123, Some::class]` would become `[Some::class]` - Obviously `123` is invalid, but are you supressing otherwise noisy errors for invalid input?

Not suggesting this _must_ be changed, but it's worth considering the implications. Might be easier to just return the un-modified input

Reply:
This explicitly filters out valid provides I mentioned exist.

Reply:
Modified method, so that it does not modify the providers.